bpo-41045: Document debug feature of f-strings ('=')

amaajemyfren opened the pull request:
Co-Authored-By: Rishi <rishi93dev@gmail.com>

<!-- issue-number: [bpo-41045](https://bugs.python.org/issue41045) -->
https://bugs.python.org/issue41045
<!-- /issue-number -->


Automerge-Triggered-By: @gvanrossum

amaajemyfren committed:
Update lexical_analysis.rst

This is a trial at closing bpo-41045.

amaajemyfren committed:
Update lexical_analysis.rst

amaajemyfren committed:
Update lexical_analysis.rst

gvanrossum commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -671,8 +672,9 @@ The parts of the string outside curly braces are treated literally,
-Python expression.  After the expression, there may be a conversion field,
-introduced by an exclamation point ``'!'``.  A format specifier may also
+Python expression. To display both the expression text and its value after evaluation,
+(useful in debugging), an equals ``'='`` may be added after the expression. A conversion field,
```
"equals" -> "equal sign" everywhere (except in the index entries, since it's spelled "equals" in other index entries for the same character).

Also please reflow the lines you changed to stay within ~80 characters.

gvanrossum commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -690,6 +692,16 @@ left to right.
+expression unless a conversion ``'!r'`` is declared.
+
+.. versionadded:: 3.8
+   The equals ``'='`` was added in Python 3.8.
```
"equal sign"

gvanrossum reviewed (commented):
Thanks! I have two small requests.

amaajemyfren committed:
Update lexical_analysis.rst

Change "equals" → "equal sign"
Limit line length to 80 characters with the exception of productionList

amaajemyfren committed:
Modified lexical_analysis.rst

Changed the index change = from equal to equals

amaajemyfren committed:
Update lexical_analysis.rst

Add index entry for symbols

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

amaajemyfren committed:
Update lexical_analysis.rst

Remove extra spaces.

gvanrossum reviewed (approved):
Okay, I'll merge this. Thanks!
